src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseExportService.java

```diff
@@ -61,18 +63,25 @@
 
     private final GitService gitService;
 
-    public ProgrammingExerciseExportService(ProgrammingExerciseRepository programmingExerciseRepository, FileService fileService, GitService gitService) {
+    private final Optional<VersionControlService> versionControlService;
+
+    private final ZipFileService zipFileService;
+
+    public ProgrammingExerciseExportService(ProgrammingExerciseRepository programmingExerciseRepository, FileService fileService, GitService gitService,
+            Optional<VersionControlService> versionControlService, ZipFileService zipFileService) {
```

Comment:
Codacy found an issue: [Parameter 'versionControlService' is not assigned and could be declared final](https://app.codacy.com/gh/ls1intum/Artemis/pullRequest?prid=5968852)